### PR TITLE
Feature/collections

### LINKED ISSUE #4: MVP - See a collection of favourite albums
As a user, I should see a navtab of collections, after clicking the collection tab I should see a list of my favourite albums, if there are no favourite albums inform the user that there are no saved albums in the collection

User implications:
1.Easy Navigation: Users can easily access their favorite albums through the "Collections" tab in the navigation menu. It provides a straightforward and intuitive way to find and manage their saved albums.

2.Quick Access to Favorites: With the collections tab, users can quickly view their favorite albums without the need to search or scroll through the entire collection of albums.

System implications:
1.Navtab Implementation: The system needs to implement the navigation menu with a "Collections" tab that users can interact with to access their favorite albums. This could involve designing the user interface and integrating the necessary frontend components.

### PR DESCRIPTION
**Pull request by jalbe0076**
This PR will close #4
## What I did:
- Refactored code for readability & removed duplicate methods
- Refactored CSS for responsiveness and additional minor styling
- Expanded mock user data
- Utilized the card component to render the users collection
- Created a sad path if the user does not have any collections saved
- Tested user flow using Cypress for happy and sad tests

## Testing Notes/Extras: 
- The user should be able to see a list of albums when they click the collections tab
- The user should see a message informing them if they do not have any albums saved in their collection
- Note the comment in the API call file informing devs that the `.then` in the fetch master_id endpoint will need to be pulled out of the function

Did this break anything?

- [X]  No
- [ ]  Yes

Type of change

- [X]  New feature (non-breaking change which adds functionality)
- [ ]  Bug fix (non-breaking change which fixes an issue)
- [X]  Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ]  Super small fix (Corrected a typo, removed a comment, etc.)
- [ ]  Skip all the other stuff and briefly explain the fix.

Checklist:

- [X]  If this code needs to be tested, all tests are passing
- [X]  The code runs locally
- [X]  There are comments where clarification/ organization is needed
- [X]  The code is DRY. If it's not, I tried my best
- [X]  I reviewed my code before pushing

What's next?
- Discover | Journal | Logout user + visuals | Context API

### CODE REVIEW

File: src/components/Card/Card.css [hunk 1]
```diff
@@ -11,9 +10,7 @@
 }
 
 .results--image {
-  height: 130px;
-  width: 130px;
-  margin-top: 40px;
```
Comment (alivaditis) [inline src/components/Card/Card.css:16]: I personally think the size of the imgs here should be hardcoded, accounts for weird sizings and extra pixelazation.  Getting rid of the media queries for grid columns was a good call tho.

**Review by alivaditis — COMMENTED** [review]
Great job getting the collections page up.  I have one css nitpick, the grid img would look better if it is always the same size.

File: src/components/Card/Card.css (hunk 1 above)
Comment (jalbe0076) [reply src/components/Card/Card.css:16]: I agree with the pixel issue. But I feel that hard-coding them made the space between them really wide. I think we should take a look and make a decision. If we hard code it, it should be in the grid sizing. I found 161px to be the best size and not being pixilated. 
Maybe we can find a way to limit the size to between 130-161px and they only adjust within those dims?

Comment (alivaditis) [reply src/components/Card/Card.css:16]: Cool yeah, we can talk about it and revisit it in general styling updates.

**Review by alivaditis — APPROVED** [review]
great job!

File: src/api.js [hunk 2]
```diff
@@ -1,60 +1,44 @@
...
-          handleError(res)
-        })
-        .then(data => fetch(`https://api.discogs.com/database/search?type=master&format=vinyl&key=GimREdkHlKcSjALMSwEP&secret=RZbpExNDRyTdbTAaiVxiJpiYgOcydrMJ&page=1&per_page=5&genre=${data.genres[0]}`))
-        .then(res => {
-          if (res.ok) {
-            return res.json()
-          }
-          handleError(res)
-        })
-  }
```
Comment (mbenfowler) [inline src/api.js:31]: Can we leave this in because I think I still need to consume it for my ticket?

File: src/components/Recommended/Recommended.js [hunk 3]
```diff
@@ -2,10 +2,10 @@ import React from 'react'
 import RecommendedAlbum from '../RecommendedAlbum/RecommendedAlbum'
 import './Recommended.css'
 
-const Recommended = ({ recommendedData }) => {
+const Recommended = ({ trendingData }) => {
     
-    if (recommendedData) {
-        const recommendedAlbums = recommendedData.results.map(item => <RecommendedAlbum key={item.id} id={item.id} title={item.title} coverImg={item.cover_image}/>)
+    if (trendingData) {
+        const recommendedAlbums = trendingData.results.map(item => <RecommendedAlbum key={item.id} id={item.id} title={item.title} coverImg={item.cover_image}/>)
```
Comment (mbenfowler) [inline src/components/Recommended/Recommended.js:8]: I gotta ask why you changed this to 'trending' - I think when I designed it I was thinking this component could be reused for "Trending"(`/trending`) AND "Recommended for you" (`/discover`) albums.  I think `recommendedData` would best to accommodate both types?

File: src/components/App/App.css [hunk 4]
```diff
@@ -6,16 +6,46 @@
...
 
 body {
   margin: auto;
   max-width: 1200px;
+  min-height: 100vh;
 }
 
 main {
-    display: flex;
-    justify-content: center;
```
Comment (mbenfowler) [inline src/components/App/App.css:18]: looks like this change disrupted the formatting, mainly on `/trending` but the placeholders on the tab pages are also no longer centered.  Why was this removed?
Screenshot 2023-07-27 at 7 13 22 PM

File: cypress/e2e/collections-spec.cy.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    cy.visit('http://localhost:3000/trending')
+  })
+
+  it('should have a saved collection', () => {
+    cy.get('.banner-container').contains('h1', 'SOUND STASH')
+      .get('.banner-container').contains('a', 'DISCOVER')
+      .get('.recommendedContainer').find('.link').should('have.length', 5).first().find('img').should('have.attr', 'src')
+      .get('.recommendedContainer').find('.link').first().find('.album').contains('p', 'Queens')
+      .get('.recommendedContainer').find('.link').last().find('img').should('have.attr', 'src')
+      .get('.recommendedContainer').find('.link').last().find('.album').contains('p', 'Akira')
```
Comment (mbenfowler) [inline cypress/e2e/collections-spec.cy.js:19]: curious as to why you're testing these in the collections spec?

**Review by mbenfowler — CHANGES_REQUESTED** [review]
Just needing some info really

File: src/api.js (hunk 2 above)
Comment (jalbe0076) [reply src/api.js:31]: You can still use it, it was just renamed and you can add what you have on line 24 & 25 after calling the function. 3 parts of the app need what we're fetching on line 17, so I think it makes more sense to take out line 24 & 25 and just use it when you invoke the function.

File: src/components/Recommended/Recommended.js (hunk 3 above)
Comment (jalbe0076) [reply src/components/Recommended/Recommended.js:8]: I changed it because I thought we could improve the app by only fetching the trending data on visiting the page once, and just keeping it in state since this data would never change in the short term. If we change it to how you described it we would need to fetch it every time we visit `/trending` and change the state. The `discover` feature would be random, so this would need to fetch all the time. We can talk about how to proceed, I don't mind changing it back if it makes sense.

File: src/components/App/App.css (hunk 4 above)
Comment (jalbe0076) [reply src/components/App/App.css:18]: yeah, something went off here, but if should be an easy fix. I didn't focus on this since we have a ticket for the styling. Also, I don't think the tabs were ever perfectly centred, but it might shift when we add a login and user icon, so I didn't fix it.

File: src/api.js (hunk 2 above)
Comment (mbenfowler) [reply src/api.js:31]: ah okay i see now thanks!

File: cypress/e2e/collections-spec.cy.js (hunk 5 above)
Comment (jalbe0076) [reply cypress/e2e/collections-spec.cy.js:19]: Probably a good idea to move it to a trending file... I'll do that as well as add sad paths for 404 & 500 level errors.

Comment (mbenfowler) [reply cypress/e2e/collections-spec.cy.js:19]: we gonna fix this up in a follow-on for polishing testing
